mkt/account/tests/test_views.py

```diff
@@ -835,3 +836,87 @@ def test_signup_about_apps(self, subscribe):
             self.VALID_EMAIL, 'mozilla-and-you,marketplace-desktop',
             lang='en-US', country='', trigger_welcome='Y',
             optin='Y', format='H')
+
+
+class TestGroupsViewSet(RestOAuth):
+    fixtures = fixture('user_2519', 'user_999')
+
+    def setUp(self):
+        super(TestGroupsViewSet, self).setUp()
+        self.grant_permission(self.user, 'Admin:%')
+
+        self.targetUser = UserProfile.objects.get(pk=999)
+        self.normal_group = Group.objects.create(name="NGroup", rules="")
+        self.restricted_group = Group.objects.create(name="RGroup", rules="",
+                                                     restricted=True)
+        self.url = reverse('account-groups', kwargs={'pk': 999})
+
+    def test_verbs(self):
+        self._allowed_verbs(self.url, ('get', 'delete', 'post'))
+
+    def test_anon(self):
+        eq_(self.anon.get(self.url).status_code, 403)
+
+    def test_non_admin(self):
+        self.remove_permission(self.user, 'Admin:%')
+        eq_(self.client.get(self.url).status_code, 403)
+
+    def test_list(self):
+        GroupUser.objects.create(group=self.normal_group,
+                                 user=self.targetUser)
+        GroupUser.objects.create(group=self.restricted_group,
+                                 user=self.targetUser)
+        res = self.client.get(self.url)
+        eq_(res.status_code, 200, res.content)
+        data = json.loads(res.content)
+        # check target has those two groups
+        eq_(data[0]['id'], self.normal_group.pk)
+        eq_(data[0]['name'], self.normal_group.name)
+        eq_(data[0]['restricted'], self.normal_group.restricted)
+        eq_(data[1]['id'], self.restricted_group.pk)
+        eq_(data[1]['name'], self.restricted_group.name)
+        eq_(data[1]['restricted'], self.restricted_group.restricted)
+
+    def do_post(self, group_id):
+        return self.client.post(self.url, data='{"group":"%s"}' % group_id)
+
+    def do_delete(self, group_id):
+        return self.client.delete(self.url, data={'group': group_id})
```

Comment:
Inconsistent, here you are not even sending JSON but a regular parameter (which we support, too, but you should pick one)